source/globals.d.ts

```diff
@@ -1,6 +1,20 @@
-declare const browser: AnyObject;
+type ArgumentTypes<TFunction> = TFunction extends (...args: infer TInferredArgs) => infer TInferredReturnType ? TInferredArgs : never;
 
-type AnyObject = {[key: string]: any};
+type FirstArg<TFunction extends any[]> =
+	TFunction extends [infer TFirstArg, ...any[]] ? TFirstArg :
+		TFunction extends [] ? undefined :
+			never;
+
+type TailArgs<TFunction extends any[]> = ((...args: TFunction) => any) extends ((
+	_: infer First,
+	...rest: infer Rest
+) => any)
+	? TFunction extends any[] ? Rest : ReadonlyArray<Rest[number]>
+	: [];
+
+declare const browser: typeof chrome;
```

Comment:
Typed the browser variable properly.

Reply:
Yup

Reply:
We're using `browser` because it's `Promise`d. We can't whitelist the plain `chrome` object.

Reply:
Back to just the `browser` property. I'll build the type for browser. Not much of the API is used in the project, so shouldn't be too painful.